src/Server.php

```diff
@@ -226,7 +229,24 @@ public function handleRequest(ConnectionInterface $conn, RequestInterface $reque
         );
 
         $callback = $this->callback;
-        $callback($request, $response);
+
+        $promise = $callback($request);
+        if (!$promise instanceof Promise) {
+            $promise = new Promise(function ($resolve, $reject) use ($promise) {
+                $resolve($promise);
+            });
+        }
```

Comment:
These 6 lines can be shortened to:

```php
$promise = \React\Promise\resolve($callback($request));
```

Additional benefit: supports also foreign thenables.

Another question: What happens if `$callback` synchronously throws an exception/throwable? This should probably turned into a rejected promises:

```php
try {
    $promise = \React\Promise\resolve($callback($request));
} catch (\Throwable $e) {
    $promise = \React\Promise\reject($e);
} catch (\Exception $e) {
    $promise = \React\Promise\reject($e);
}
```

I'd prefer this over just letting the exception bubble up because it would handle exceptions in the same ways as if `$callback` returns a rejected promises (see also my next comment).